Submitted By: Robert Connolly <robert at linuxfromscratch dot org> (ashes)
Date: 2004-11-07
Initial Package Version: 5.8.5
Upstream Status: Submitted
Origin: http://www.uclibc.org/lists/uclibc/2004-January/007939.html
Description: Perl's fast-standard-input/output code is not portable.

Also see:
http://www.mail-archive.com/perl5-porters%40perl.org/msg81844.html

Also see:
http://www.uclibc.org/
http://www.linuxfromscratch.org/hlfs/

diff -Naur perl-5.8.5.orig/perlio.c perl-5.8.5.uclibc/perlio.c
--- perl-5.8.5.orig/perlio.c	2004-03-16 18:36:08.000000000 +0000
+++ perl-5.8.5.uclibc/perlio.c	2004-11-07 18:36:05.861324248 +0000
@@ -2824,7 +2824,11 @@
     /* XXX this could use PerlIO_canset_fileno() and
      * PerlIO_set_fileno() support from Configure
      */
-#  if defined(__GLIBC__)
+#  if defined(__UCLIBC__)
+    /* uClibc must come before glibc because it defines __GLIBC__ as well. */
+    f->__filedes = -1;
+    return 1;
+#  elif defined(__GLIBC__)
     /* There may be a better way for GLIBC:
     	- libio.h defines a flag to not close() on cleanup
      */	
